app/controls.tsx

```diff
@@ -1,6 +1,45 @@
+import { useEffect, useState } from "react";
 import Select from "react-select";
 
-const Controls = () => {
+
+interface ControlsProps {
+  onSortInput:(value:{field: string,direction: string})=>void
+}
+
+const Controls: React.FC<ControlsProps> = ({onSortInput}) => {
+
+  const [sortingOptions, setSortingOptions] = useState({
+    field: "name",
+    direction: "ascending",
+  });
+
+  /**
+   * Updates the sorting options with the new field value.
+   *
+   * @param {Object} param - An object containing the new field value.
+   * @param {any} param.value - The new field value to set.
+   * @return {void} This function does not return anything.
+   */
+  const handleSortInput = ({value}:any): void => {
+    setSortingOptions({ ...sortingOptions, field:value });
+  };
+
+  /**
+   * Set the sorting options with the new direction value.
+   *
+   * @param {any} value - The new direction value to set.
+   */
+  const handleDirectionInput = ({value}:any) => {
+    setSortingOptions({ ...sortingOptions, direction:value });
+  };
+
+  // useEffect that would send onSortInput
+  useEffect(() => {
+    if(sortingOptions) {
+      onSortInput(sortingOptions);
```

Comment:
Code above triggers this `useEffect` that would send a callback to `onSortInput` which gives the parent component the ability to react to this effect.